system/src/main.cpp

```diff
@@ -415,6 +415,8 @@ extern "C" void HAL_SysTick_Handler(void)
     }
 #endif
 
+    BUTTON_Timer_Handler();
```

Comment:
Please add stub `BUTTON_Timer_Handler()` for all the other platforms.

Reply:
Can I use `if PLATFORM_ID == PLATFORM_XENON || PLATFORM_ID == PLATFORM_ARGON || PLATFORM_ID == PLATFORM_BORON` here?

If I add stub `BUTTON_Timer_Handler()`, I just need to add it to below file, right?
```
platform/MCU/STM32F1xx/SPARK_Firmware_Driver/inc/hw_config.h
platform/MCU/STM32F1xx/SPARK_Firmware_Driver/inc/hw_config.c
platform/MCU/STM32F2xx/SPARK_Firmware_Driver/inc/hw_config.h
platform/MCU/STM32F2xx/SPARK_Firmware_Driver/inc/hw_config.c
```

Reply:
Or even better let's add a new feature flag to `hal_platform.h` and set it in Gen 3-specific `hal_platform_config.h` and `#if` on it. Let's say `HAL_PLATFORM_BUTTON_DEBOUNCE_IN_SYSTICK`?

Reply:
Sounds good, better than adding stub function to other platforms. I'll add `HAL_PLATFORM_BUTTON_DEBOUNCE_IN_SYSTICK` to `hal/src/nRF52840/hal_platform_nrf52840_config.h`